### ISSUE TITLE
ChromiumOS + upstream kernel 5.10.x stable : boot issues (modules compression)

### ISSUE THREAD

**Issue by nuclearcat** [kernelci/kernelci-core#1246]
During tests with kernel 5.10.x i noticed that Octopus Chromebook and QEMU behaving erratically, e.g. they randomly reboot, networking doesnt work, and tast tests are not being executed.

To reproduce issue i run manually on my QEMU one of 5.10.x released built by kernelci and did some troubleshooting over QEMU serial console. During that i noticed that none of kernel modules loaded and any attempt to load them ends with failure.

During troubleshooting modprobe / insmod with strace, i noticed difference between working kernel (5.15) and 5.10 was related to modules compression:
```
openat(AT_FDCWD, "/sys/module/compression", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
```

As experiment i built kernel without modules compression and it worked, so problem is definitely related to modules compression.

But issue is a bit complex:

Primary reason, 5.10 kernel and ChromeOS userspace (kmod and busybox modprobe/insmod) doesnt play well with modules compression PROBABLY due multiple reasons:

1)It interferes with loadpin LSM enabled and enforced
But even if we disable loadpin enforcement:
2)kmod tools in ChromeOS (kmod version 25) seems doesnt handle well dependencies format of modules generated by kernelci (kmod version 28):

localhost ~ # ls /lib/modules/5.10.120/kernel/net/ipv4/tcp_lp.ko.gz
/lib/modules/5.10.120/kernel/net/ipv4/tcp_lp.ko.gz

localhost ~ # modprobe tcp_lp
modprobe: FATAL: Module tcp_lp not found in directory /lib/modules/5.10.120

localhost ~ # insmod /lib/modules/5.10.120/kernel/net/ipv4/tcp_lp.ko.gz
localhost ~ #

localhost ~ # lsmod |grep tcp_lp
tcp_lp 16384 0

What is more interesting (sure removed module again):
localhost ~ # busybox modprobe tcp_lp
localhost ~ #

So busybox modprobe works, kmod userspace in chromeos doesnt!

And last, most interesting part:
localhost ~ # mount -o remount,rw /
localhost ~ # depmod -a
localhost ~ # lsmod |grep tcp_lp
tcp_lp 16384 0

I'm not really sure what can be done next, as several ways of solving this issue is possible.

As i see ChromeOS team did backported to their downstream 5.10.x in-kernel modules decompression which is likely solving issue (i still need to test their patch):
https://chromium.googlesource.com/chromiumos/third_party/kernel/+log/refs/heads/chromeos-5.10/kernel/module_decompress.c

**Comment by nuclearcat** [comment]
LoadPin enforcement issue is solved by patch cec969b25df77d5bdd09676adbac1094390c683b
https://chromium.googlesource.com/chromiumos/third_party/kernel/+/cec969b25df77d5bdd09676adbac1094390c683b%5E%21/

After that i can load modules by insmod, but not modprobe.

**Comment by nuclearcat** [comment]
Here is one test example in dashboard:
https://chromeos.kernelci.org/test/plan/id/62997d1a99cf5a59747df419/

Original test re-run (because old in dashboard was not detecting prompt properly):
https://lava.collabora.co.uk/scheduler/job/6562829
But we can see specific fault pattern, when kernel cannot load modules properly, many upstart processes are crashing and restarting:
<12>[    6.842449] init: wpasupplicant main process (926) terminated with status 255
<12>[    6.850900] init: wpasupplicant main process ended, respawning
<12>[    6.874422] init: wpasupplicant main process (941) terminated with status 255
<12>[    6.883213] init: wpasupplicant main process ended, respawning
<12>[    6.906549] init: wpasupplicant main process (949) terminated with status 255

And here is job result with modules decompressed:
https://lava.collabora.co.uk/scheduler/job/6529760
As we can see it even passing tast tests

**Comment by nuclearcat** [comment]
Yes, same issue observed on QEMU.

Denverton might be not correct config for octopus, but still usable and improvement visible (basic tast tests are passing).
Octopus is Intel Geminilake, we dont have such fragment.

**Comment by gctucker** [comment]
I think it's worth checking in the Chrome OS source tree which fragments are used when the build is set up for `octopus`.

**Comment by nuclearcat** [comment]
Issue addressed in https://github.com/kernelci/kernelci-core/pull/1318 and https://github.com/kernelci/kernelci-core/pull/1306
Results of tests show that it solve issue

**Comment by gctucker** [comment]
> Issue addressed in #1318 and #1306 Results of tests show that it solve issue

Please link the PRs in the "development" section of this issue on the right hand side.

This also happens automatically by mentioning the issue when creating a new PR (e.g. "Fixed #1318")

**Comment by nuclearcat** [comment]
> > Issue addressed in #1318 and #1306 Results of tests show that it solve issue
> 
> Please link the PRs in the "development" section of this issue on the right hand side.
> 
> This also happens automatically by mentioning the issue when creating a new PR (e.g. "Fixed #1318")

Thanks, i was not aware about such feature.
